### DISCUSSION TITLE
Session not getting created after google/facebook oAUTH

### DISCUSSION THREAD

**Discussion by varun86b4u** [supabase/supabase#17646] [Questions] [answered]
The session was working 2 days back, there is no change in the code. 

code used:

const { data, error } = await supabase.auth.signInWithOAuth({
      provider: "google",
      options: {
        queryParams: {
          access_type: "offline",
          prompt: "consent",
        }
      },
    })

const { data, error } = await supabase.auth.signInWithOAuth({
      provider: "facebook"
    })
    
    its redirects fine after google/facebook oauth, but session is empty.
    
    Supabase version : 1.93.0

**Comment by GaryAustin1** [comment]
Do you await getSession() after doing createClient()?

**Reply by varun86b4u** [reply]
Yes,

const supabase = createMiddlewareClient({ req, res })
const { data, error } = await supabase.auth.getSession()

My observation is cookies are not getting set, after oauth redirection.

**Reply by varun86b4u** [reply]
https://www.dev.snaptales.ai/registration

It can be tried out.. google auth.. 
it redirects back to registration due to session not found and cookies are not set.

**Comment by varun86b4u** [comment, marked as answer]
Got the issue, looks like its going on PKCE flow

https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-proof-key-for-code-exchange-pkce

Adding below in callback URL

 const code = req.query.code

  if (typeof code === 'string') {
    // exchange the auth code for user session
    await supabase.auth.exchangeCodeForSession(code)
  }
